examples/super_resolution/main.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchvision
+import torchvision.transforms as transforms
+from model import Net
+from torch.utils.data import DataLoader
+
+from ignite.engine import Engine, Events
+from ignite.metrics import PSNR
+
+# Training settings
+parser = argparse.ArgumentParser(description="PyTorch Super Res Example")
+parser.add_argument("--upscale_factor", type=int, required=True, help="super resolution upscale factor")
+parser.add_argument("--batchSize", type=int, default=64, help="training batch size")
+parser.add_argument("--testBatchSize", type=int, default=10, help="testing batch size")
+parser.add_argument("--nEpochs", type=int, default=2, help="number of epochs to train for")
+parser.add_argument("--lr", type=float, default=0.01, help="Learning Rate. Default=0.01")
+parser.add_argument("--cuda", action="store_true", help="use cuda?")
+parser.add_argument("--mps", action="store_true", default=False, help="enables macOS GPU training")
+parser.add_argument("--threads", type=int, default=4, help="number of threads for data loader to use")
+parser.add_argument("--seed", type=int, default=123, help="random seed to use. Default=123")
+opt = parser.parse_args()
+
+print(opt)
+
+if opt.cuda and not torch.cuda.is_available():
+    raise Exception("No GPU found, please run without --cuda")
+if not opt.mps and torch.backends.mps.is_available():
+    raise Exception("Found mps device, please run with --mps to enable macOS GPU")
+
+torch.manual_seed(opt.seed)
+use_mps = opt.mps and torch.backends.mps.is_available()
+
+if opt.cuda:
+    device = torch.device("cuda")
+elif use_mps:
+    device = torch.device("mps")
+else:
+    device = torch.device("cpu")
+
+print("===> Loading datasets")
+
+
+class SRDataset(torch.utils.data.Dataset):
+    def __init__(self, dataset, scale_factor):
+        self.dataset = dataset
+        self.transform = transforms.Resize(
+            (len(dataset[0][0][0]) * scale_factor, len(dataset[0][0][0][0]) * scale_factor)
+        )
```

Comment:
I think you can just use torchvision functional API instead of this:
```
from torchvision.transforms.functional import resize
```

Reply:
So, looks like you are upsampling input image with bilinear mode. Model will learn how to perform bilinear upsampling which is not super-resolution...

I think my previous point about CIFAR10 was about using it for evaluation only and not training. We should train on large images rescaled to small ones (as dataset you can pick one among available, scenes/classes similar to CIFAR10 and downloadable on torchvision: CalTech101, VOC, etc, https://pytorch.org/vision/stable/datasets.html) 

Then we can visually check generalization on CIFAR10.

Reply:
Hey @vfdev-5 I made changes to get the images rescaled to smaller ones during training as present in the `PyTorch/examples` and suggested by you.